includes/class-edd-customer.php

```diff
@@ -333,6 +333,10 @@ public function remove_payment( $payment_id = 0, $update_stats = true ) {
 			return false;
 		}
 
+		if ( 'publish' !== edd_get_payment_status( $payment_id, false ) ) {
```

Comment:
`revoked` should be used here since that is a completed payment (not reversed) that has simply been disabled.